### PR TITLE
6917 credit card transactions via credit card tokens

### PR DESCRIPTION
**Pull request by fraserd**
@nothinhappened , @cattiwill , @dandersonpeloton :  I have made this new PR which has all of my pending changes (including those from PR #1 ).

Please review these changes and give your +1 or -1 in a comment.

I will merge/make changes once all reviewers have had an opportunity to give feedback.

Thanks!

### CODE REVIEW

File: PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        string OrderNumber { get; set; }
+        string CardOwner { get; set; }
+
+        CreditCardNumber CardNumber { get; set; }
+
+        string CardType { get; }
+        string ExpiryMonth { get; set; }
+        string ExpiryYear { get; set; }
+        string CardSecurityCode { get; set; }
+        decimal? CardLast4Digits { get; }
```
Comment (cattiwill) [inline PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs]: Remove as provided by CreditCardNumber class

File: PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        string MaskedCardDigits { get; }
```
Comment (cattiwill) [inline PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs]: Remove as provided by CreditCardNumber class

File: PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+namespace PelotonEppSdk.Interfaces
+{
+    public interface ICreditCardUpdateRequest
+    {
+        string OrderNumber { get; set; }
+        string CardOwner { get; set; }
+
+        CreditCardNumber CardNumber { get; set; }
+
+        string CardType { get; }
```
Comment (cattiwill) [inline PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs]: Remove as provided by CreditCardNumber class

File: PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        string ExpiryMonth { get; set; }
+        string ExpiryYear { get; set; }
+        string CardSecurityCode { get; set; }
```
Comment (cattiwill) [inline PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs]: If we are not verifying the card on the update request thne we don't need CardSecurityCode to be sent

File: PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+using PelotonEppSdk.Models;
+
+namespace PelotonEppSdk.Interfaces
+{
+    public interface ICreditCardUpdateRequest
+    {
+        string OrderNumber { get; set; }
+        string CardOwner { get; set; }
+
+        CreditCardNumber CardNumber { get; set; }
```
Comment (cattiwill) [inline PelotonEppSdk/Interfaces/ICreditCardUpdateRequest.cs]: We are not going to allow updates to CardNumbers. A create credit card request will need to be sent when changing the card number.

File: PelotonEppSdk/Models/Address.cs [hunk 6]
```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics.CodeAnalysis;
+
+namespace PelotonEppSdk.Models
+{
+    public class Address
+    {
+//        [Display(ResourceType = typeof(Resources.Res), Name = "Address")]
```
Comment (cattiwill) [inline PelotonEppSdk/Models/Address.cs]: Delete Display DataAnnotations. Remove globalization from constraint DataAnnotations.

File: PelotonEppSdk/Models/Address.cs [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        }
+    }
+
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    internal class address
+    {
+        /// <summary>
+        /// Address line 1
+        /// </summary>
+        [StringLength(256)]
```
Comment (cattiwill) [inline PelotonEppSdk/Models/Address.cs]: Move constraint DataAnnotations up to public class

File: PelotonEppSdk/Models/BankAccount.cs [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+        [Required, RegularExpression("CAD|USD")]
+        public string CurrencyCode { get; set; }
+ }
+
+    /// <summary>
+    /// The set of properties which define a bank account
+    /// </summary>
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
+    internal class bank_account
```
Comment (cattiwill) [inline PelotonEppSdk/Models/BankAccount.cs]: Summary Info and DataAnnotation need to be promoted to public class

**Comment by nothinhappened** [comment]
+1, please wait for executive approval before merging

**Pull request merged**